### PR TITLE
Improve JSON output from ddev version

### PR DESCRIPTION
**Pull request by cweagans**
## The Problem/Issue/Bug:

The JSON output from `ddev version -j` can't be parsed easily.

## How this PR Solves The Problem:

We're using the new output system to determine how to render the data (JSON or ASCII table).

I also moved the GetVersionInfo() function and it's corresponding test to the version package, since it makes a bit more sense for it to live there.

## Manual Testing Instructions:

Run `ddev version` and `ddev version -j`. Make sure the former works as it did before, and the latter should be a JSON object with individual properties for each key.

## Automated Testing Overview:

The existing test was repurposed to test `version.GetVersionInfo()`. I also added a new test to look at the output of `ddev version` directly.

## Related Issue Link(s):

https://github.com/drud/ddev/issues/568

### CODE REVIEW

**Comment by cweagans** [comment]
The fails on the last test run didn't seem related to this PR. All the tests passed on Windows of all places.

**Comment by rfay** [comment]
It's good old broken darwin again. Or broken docker-on-darwin, not sure which. I'll run it manually later.

**Comment by rfay** [comment]
So my fix must not have been successful. This is the fail you're getting on surf-darwin (although the gist is somehow horribly corrupted:

--- FAIL: TestListWithoutDir (30.33s)
        Error Trace:    ddevapp_test.go:771
	Error:		Expect "NAME  TYPE     LOCATION                                                                                                                         URL(s)                   STATUS
			junk  drupal7  /private/var/folders/7m/133fr9vn6b74srxz5t6t7bs00000gn/T/t-ddev-7f65df-2018-01-13T05.02.27Z/TestStartWithoutDdevConfig376700240  http://junk.ddev.local   app directory missing:
			                                                                                                                                                https://junk.ddev.local  /private/var/folders/7m/133fr9vn6b74srxz5t6t7bs00000gn/T/t-ddev-7f65df-2018-01-13T05.02.27Z/TestStartWithoutDdevConfig376700240             " to match "app directory missing.*/private/var/folders/7m/133fr9vn6b74srxz5t6t7bs00000gn/T/t-ddev-7f65df-2018-01-13T05.02.27Z/TestStartWithoutDdevConfig376700240"

File: pkg/version/version_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"testing"
+
+	asrt "github.com/stretchr/testify/assert"
+)
+
+func TestGetVersionInfo(t *testing.T) {
+	assert := asrt.New(t)
+	v := GetVersionInfo()
+
+	assert.Contains(v, "cli")
```
Comment (rfay) [inline pkg/version/version_test.go]: Aren't all these first items redundant? The assertions below will fail if the keys don't exist right?

Comment (cweagans) [reply pkg/version/version_test.go]: Probably. I wrote the first items, and then thought better of it and actually checked the values. The first tests are just leftovers. I'll remove them.

**Comment by cweagans** [comment]
Rebased after #602

**Comment by rfay** [comment]
Yay!

**Comment by rickmanelius** [comment]
Ready to merge?

**Comment by cweagans** [comment]
Not just yet, I think.

> How did build-info get to the top of the list? It's the least likely thing to be valuable, so should be at the bottom IMO.

Map ordering is indeterminate, so I'm not sure if we can control this.

> Could we wrap the build info? I'm pretty sure the long length of build info causes the strange tabular output with a line between each item. If it's something else, it would be nice to solve while we're here. Anyway, the regular non-json ddev version doesn't need to have a line between each item.

In a window that's wide enough to accommodate everything, it looks like this:  http://take.ms/I8Sdt
If it's too narrow, it looks like this: http://take.ms/ek2VP

I'm not really sure how to solve this, as the build info is generated somewhere else, right? We're just using what we're given.

**Comment by cweagans** [comment]
What if we make build info just contain the git hash? It's always gonna be built with the build container, so we don't need to include that. Since it's built in a container, we probably don't care too much what specific host it was built on. Maybe just commit + build date or something?

**Comment by rfay** [comment]
We already have a good committish, which is good enough.

We've never used the build-info for anything ever. 

Why don't we just drop it from both json and tabular versions? The other alternative would be to remove it from the table and output it below.

**Comment by rfay** [comment]
Also, I don't want this to block this PR, so my slight suggestion of "Why don't we solve this other thing" can just be ignored.

**Comment by cweagans** [comment]
Let's drop it.

**Comment by rfay** [comment]
So nice and compact now! 
1__default__bash_

Wish the ordering were reasonable, but I can live with this for now.

Also, build info is still there in ddev version -j, so it's easy to access if we want it.

**Pull request merged**